compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/lir/aarch64/AArch64ArrayFillOp.java

```diff
@@ -0,0 +1,338 @@
+import static jdk.vm.ci.aarch64.AArch64.r5;
+import static jdk.vm.ci.aarch64.AArch64.r6;
+import static jdk.vm.ci.aarch64.AArch64.r7;
+import static jdk.vm.ci.aarch64.AArch64.r8;
+import static jdk.vm.ci.aarch64.AArch64.r9;
+import static jdk.vm.ci.code.ValueUtil.asRegister;
+
+import jdk.graal.compiler.asm.Label;
+import jdk.graal.compiler.asm.aarch64.AArch64Address;
+import jdk.graal.compiler.asm.aarch64.AArch64Assembler.ConditionFlag;
+import jdk.graal.compiler.asm.aarch64.AArch64Assembler.ShiftType;
+import jdk.graal.compiler.asm.aarch64.AArch64MacroAssembler;
+import jdk.graal.compiler.debug.GraalError;
+import jdk.graal.compiler.lir.LIRInstructionClass;
+import jdk.graal.compiler.lir.Opcode;
+import jdk.graal.compiler.lir.SyncPort;
+import jdk.graal.compiler.lir.asm.CompilationResultBuilder;
+import jdk.vm.ci.aarch64.AArch64Kind;
+import jdk.vm.ci.code.Register;
+import jdk.vm.ci.meta.JavaKind;
+import jdk.vm.ci.meta.Value;
+
+/**
+ * Emits code for setting all elements of {@link #array} with copies of {@link #value}. The value
+ * must be a primitive integer, and its bit size must match the array's {@link #elementType}.
+ * Filling of floating-point arrays is supported by handling the floating-point value as plain
+ * values. The assembly code in this intrinsic was based in the HotSpot's version of the same
+ * intrinsic.
+ */
+@Opcode("ARRAYS_FILL")
+// @formatter:off
+@SyncPort(from = "https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp#L2411-L2548",
+          sha1 = "0a5ef827b06ca9fc6a7cbc059e4331cfd3cd2e62")
+@SyncPort(from = "https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp#L6281-L6351",
+          sha1 = "0a5ef827b06ca9fc6a7cbc059e4331cfd3cd2e62")
+// @formatter:on
+public final class AArch64ArrayFillOp extends AArch64ComplexVectorOp {
+    public static final LIRInstructionClass<AArch64ArrayFillOp> TYPE = LIRInstructionClass.create(AArch64ArrayFillOp.class);
+
+    private JavaKind elementType;
+    @Alive({REG}) protected Value array;
+    @Alive({REG}) protected Value arrayBaseOffset;
+    @Alive({REG}) protected Value length;
+    @Alive({REG}) protected Value value;
+    @Temp protected Value[] temps;
+
+    public AArch64ArrayFillOp(JavaKind kind, Value array, Value arrayBaseOffset, Value length, Value value) {
+        super(TYPE);
+
+        GraalError.guarantee(array.getPlatformKind() == AArch64Kind.QWORD, "pointer value expected");
+        GraalError.guarantee(length.getPlatformKind() == AArch64Kind.DWORD, "integer value expected in 'length'");
+
+        this.elementType = kind;
+        this.array = array;
+        this.arrayBaseOffset = arrayBaseOffset;
+        this.length = length;
+        this.value = value;
+
+        this.temps = new Value[]{
+                        r5.asValue(),
+                        r6.asValue(),
+                        r7.asValue(),
+                        r8.asValue(),
```

Comment:
we reserve r8 and r9 as scratch registers. If you want to use then, this is is best to do something like https://github.com/oracle/graal/blob/4c7334c318922844a2aa799601322cb8b76065e7/compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/lir/aarch64/AArch64ArrayRegionCompareToOp.java#L108

Reply:
Done.